### PR TITLE
RediSearch Module

### PR DESCRIPTION
**Pull request by AvitalFineRedis**
add support to RedisSearch and test it

### CODE REVIEW

File: tests/test_search.py [hunk 1]
```diff
@@ -0,0 +1,1036 @@
...
+import time
+import os
+import sys
+
+from io import TextIOWrapper
+from .conftest import skip_ifmodversion_lt
+from redis import Redis
+
+import redisplus.search
+from redisplus.client import RedisPlus
```
Comment (AvitalFineRedis) [inline tests/test_search.py:14]: @chayim Is there a way to import all these at once?

File: tests/test_search.py [hunk 2]
```diff
@@ -0,0 +1,1036 @@
...
+    client.add_document('doc-5ghs2', play='Death of a Salesman')
+    res = client.search(Query('death of a salesman'))
+    assert (1 == res.total)
+    client.delete_document('doc-5ghs2')
+
+
+"""
+seems like this command was removed from redisearch
+https://github.com/RediSearch/RediSearch/pull/1372
+
```
Comment (AvitalFineRedis) [inline tests/test_search.py:196]: This is the command that seems to be removed from RediSearch

**Comment by AvitalFineRedis** [comment]
@chayim Can you take a look at the error that looks like a decode error?

File: tests/test_search.py (hunk 1 above)
Comment (chayim) [reply tests/test_search.py:14]: Nah, this is a nice clean way to do it.

File: tests/test_search.py (hunk 2 above)
Comment (chayim) [reply tests/test_search.py:196]: Yank it then, just double-check.

**Comment by AvitalFineRedis** [comment]
@chayim The failures are ready for you ;)

**Pull request merged**